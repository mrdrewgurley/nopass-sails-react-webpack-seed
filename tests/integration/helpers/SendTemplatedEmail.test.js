sendEmail = async () => {
  return await sails.helpers.sendTemplatedEmail.with({
    template: 'email-token-access',
    templateData: { token: 12345 }
  })
}

describe('SendTemplatedEmailHelper', () => {
  /**
   * SessionAuth#default
   * @desc Tests for the #default method.
   */
  describe('#default', () => {
    it('should match output snapshot', () => {
      sendEmail()
        .then(response => {
          response.contents.should.matchSnapshot()
        }).catch()
    })
  })
})
