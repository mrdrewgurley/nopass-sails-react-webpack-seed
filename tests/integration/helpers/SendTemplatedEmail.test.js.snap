// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendTemplatedEmailHelper #default should match output snapshot 1`] = `
"
-=-=-=-=-=-=-=-=-=-=-=-=-= Email log =-=-=-=-=-=-=-=-=-=-=-=-=-
To: user@example.com
Subject: Zero Auth Account Access

Body:

<div style=\\"background-color: #FAFAFA; width: 100%; font-family: 'Arial', 'Helvetica Neue', 'Helvetica', sans-serif; box-sizing: border-box; padding-bottom: 25px; margin: 0;\\">
  <div style=\\"background: transparent; padding-top: 30px; padding-bottom: 20px; text-align: center;\\">
    <img style=\\"display: inline-block; height: 30px; width: auto; margin-left: auto; margin-right: auto;\\" alt=\\"Logo\\" src=\\"http://via.placeholder.com/200x60/fafafa/000?text=YOUR+LOGO+HERE\\"/>
  </div>
  <div style=\\"background-color: #fff; color: #000; font-size: 1em; border-top: 2px solid #00ACC4; border-bottom: 1px solid #E6E6E6; box-sizing: border-box; padding: 25px; width: 100%; max-width: 600px; margin-left: auto; margin-right: auto;\\">

    
  <p style=\\"margin-bottom: 25px;\\">Greetings,</p>
  <p style=\\"margin-bottom: 25px;\\">You requested access to your account and can login now. Just click the button below to login.</p>
  <div style=\\"margin-bottom: 25px; text-align: center;\\">
    <a style=\\"background-color: #00ACC4; display: inline-block; font-size: 1.1em; color: #fff; padding: 10px 35px 10px; font-weight: 500; text-decoration: none; border-radius: 7px;\\"
      href=\\"http://localhost:1337/login/12345\\">Login</a>
  </div>
  <p style=\\"margin-bottom: 25px;\\">If you have any trouble, try pasting this link in your browser:
    <a style=\\"color: #00ACC4; word-wrap: break-word;\\" href=\\"http://localhost:1337/login/12345\\">
      http://localhost:1337/login/12345
    </a>
  </p>
  <p style=\\"margin-bottom: 5px;\\">Sincerely,</p>
  <p style=\\"margin-top: 0px;\\">The
    Example Organization Team</p>


  </div>
  <div style=\\"text-align: center; padding-top: 15px; margin-bottom: 15px; font-size: 12px;\\">
    <a style=\\"color: #00ACC4;\\" href=\\"http://localhost:1337/contact\\">Help</a>
    <span style=\\"color: #ccc;\\">&nbsp;&nbsp;|&nbsp;&nbsp;</span>
    <a style=\\"color: #00ACC4;\\" href=\\"http://localhost:1337/faq\\">FAQ</a>
    <span style=\\"color: #ccc;\\">&nbsp;&nbsp;|&nbsp;&nbsp;</span>
    <a style=\\"color: #00ACC4;\\" href=\\"http://localhost:1337/legal/terms\\">Terms of Service</a>
    <span style=\\"color: #ccc;\\">&nbsp;&nbsp;|&nbsp;&nbsp;</span>
    <a style=\\"color: #00ACC4;\\" href=\\"http://localhost:1337/legal/privacy\\">Privacy Policy</a>
  </div>
  <p style=\\"color: #777777; padding: 0 25px; max-width: 400px; box-sizing: border-box; margin-left: auto; margin-right: auto; font-size: 10px; text-align: center;\\">Please do not reply to this email. To get in touch with us, visit our <a style=\\"color: #00ACC4;\\" href=\\"http://localhost:1337/contact\\"> help center.</a> If you have not signed up for an account with us, please disregard this message.</p>
  <p style=\\"color: #777777; padding: 0 25px; max-width: 400px; box-sizing: border-box; margin-left: auto; margin-right: auto; font-size: 10px; text-align: center;\\"><em>(built with <a style=\\"color: #00ACC4;\\" href=\\"https://sailsjs.com\\">Sails.js</a>)</em></p>
</div>

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-"
`;
